bundles/org.openhab.binding.dwdpollenflug/src/main/java/org/openhab/binding/dwdpollenflug/internal/handler/DWDPollenflugBridgeHandler.java

```diff
@@ -0,0 +1,201 @@
+        }
+    }
+
+    private void poll() {
+        logger.debug("Polling");
+        requestRefresh().handle((pollenflug, pollException) -> {
+            if (pollenflug == null) {
+                if (pollException == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            pollException.getMessage());
+                }
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+                notifyOnUpdate(pollenflug);
+            }
+
+            return null;
+        });
+    }
+
+    private CompletableFuture<@Nullable DWDPollenflug> requestRefresh() {
+        CompletableFuture<@Nullable DWDPollenflug> f = new CompletableFuture<>();
+        Request request = client.newRequest(URI.create(DWD_URL));
+
+        request.method(HttpMethod.GET).timeout(2000, TimeUnit.SECONDS).send(new BufferingResponseListener() {
+            @NonNullByDefault({})
+            @Override
+            public void onComplete(Result result) {
+                final HttpResponse response = (HttpResponse) result.getResponse();
+                if (result.getFailure() != null) {
+                    Throwable e = result.getFailure();
+                    if (e instanceof SocketTimeoutException || e instanceof TimeoutException) {
+                        f.completeExceptionally(new DWDPollingException("Request timeout", e));
+                    } else {
+                        f.completeExceptionally(new DWDPollingException("Request failed", e));
+                    }
+                } else if (response.getStatus() != 200) {
+                    f.completeExceptionally(new DWDPollingException(getContentAsString()));
+                } else {
+                    try {
+                        DWDPollenflug pollenflugJSON = gson.fromJson(getContentAsString(), DWDPollenflug.class);
+                        f.complete(pollenflugJSON);
+                    } catch (JsonSyntaxException ex2) {
+                        f.completeExceptionally(new DWDPollingException("Parsing of response failed"));
+                    }
+                }
+            }
+        });
+
+        return f;
+    }
+
+    public synchronized boolean registerRegionListener(DWDPollenflugRegionListener regionListener) {
+        logger.debug("Register region listener");
+        boolean result = regionListeners.add(regionListener);
+        if (result) {
+            startPolling();
+        }
+        return result;
+    }
+
+    public synchronized boolean unregisterRegionListener(DWDPollenflugRegionListener regionListener) {
+        logger.debug("Unregister region listener");
+        boolean result = regionListeners.remove(regionListener);
+        return result;
+    }
+
+    public synchronized void notifyOnUpdate(@Nullable DWDPollenflug newState) {

```

Comment:
remove synchronized, this is not needed anymore

Reply:
Removed